x-pack/plugins/enterprise_search/public/applications/shared/schema/types.ts

```diff
@@ -31,7 +31,7 @@ export type Schema = Record<string, SchemaType>;
 
 // This is a mapping of schema field types ("text", "number", "geolocation", "date")
 // to the names of source engines which utilize that type
-export type SchemaConflictFieldTypes = Record<SchemaType, string[]>;
+export type SchemaConflictFieldTypes = Partial<Record<SchemaType, string[]>>;
```

Comment:
Not sure how I didn't run into a type error with this before now, but without the `Partial<>` Typescript yells if you have just e.g. `{ text: ['some-engine'], number: ['another-engine'] }` - it wants all the keys from SchemaType otherwise 🤷